### DISCUSSION TITLE
Map Language

### DISCUSSION THREAD

**Discussion by Nesimm** [organicmaps/organicmaps#662] [Ideas]
There should be an option to change the map language while in the app.  This feature is available in the OsmAnd app.  For example, I live in Turkey, the map language of the application is in Turkish, but sometimes I want to use the map in Kurdish.  I think a lot of people will use this option.  Thanks

**Comment by biodranik** [comment]
If someone contributes Kurdish translation, then you may switch the interface. There are many strings in data/strings/strings.txt file to translate (and in a few other places too).

**Comment by sanketgarade** [comment]
I also have a similar request. I don't mind the app interface language being English but I want only the map information to be displayed in a specific language that I select. 

In my case I want to see the map in Marathi language (mr). Technically speaking the display name on the map should use the `name:mr` tag, instead of the `name` tag. 

Is it possible to have such a feature?

**Reply by biodranik** [reply]
At the moment, we support a limited set of languages from OSM, and there is no Mr there :(
See https://github.com/organicmaps/organicmaps/blob/master/coding/string_utf8_multilang.cpp#L15 for more details.

**Reply by biodranik** [reply]
If you see some invalid or dead language code there, it can be replaced by another language.

**Reply by sanketgarade** [reply]
I saw few rows marked as `StringUtf8Multilang::kReservedLang`. Is it possible to use any of those for `mr`?

Among these reserved rows, I couldn't find any codes [here](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) for `hsb` and `gsw`. So I guess those codes and languages don't exist.

**Reply by biodranik** [reply]
https://wiki.openstreetmap.org/wiki/Key:name:gsw

https://taginfo.openstreetmap.org/keys/name%3Ahsb#values

You're right, we can add `mr` by reusing one of the reserved langs. Another problem could be finding a proper font to display those scripts.  Can you please check if our current font files at https://github.com/organicmaps/organicmaps/tree/master/data contain `mr` letters?

**Reply by sanketgarade** [reply]
The font `06_code2000.ttf` supports Devanagari (which is the script used by the Marathi language), but the rendering is incorrect. So it would be better to use another font. 

I suggest using [Noto Serif Devanagari](https://fonts.google.com/noto/specimen/Noto+Serif+Devanagari?subset=devanagari&category=Serif).
This not only supports Marathi `mr` but also Hindi `hi` which is also present in the supported language list you shared above.

**Reply by biodranik** [reply]
Is Hindi missing in our current fonts?

Can you please start working on the PR with these changes? We'll help :)

**Reply by sanketgarade** [reply]
> Is Hindi missing in our current fonts?

No. The language Hindi is present [here](https://github.com/organicmaps/organicmaps/blob/b63f283e7987fc9e229341940d7c54755ff230e4/coding/string_utf8_multilang.cpp#L79) while the font of its script "Devanagari" is present at `data/06_code2000.ttf`.

> Can you please start working on the PR with these changes? We'll help :)

Sure 😄 . Though I am not fully sure what all I need to do. We can take it one step at a time.
Is it ok if I open a dedicated issue for discussing this? or do you prefer to continue the discussion here?

About the PR, for now I will modify the `gsw` line with `mr`, and will add a font file for the Devanagari script.

**Reply by biodranik** [reply]
Yes, please move the discussion into PR. Also check these files in data: fonts_blacklist.txt, fonts_whitelist.txt and unicode_blocks.txt
